types/node-vault-client/index.d.ts

```diff
@@ -0,0 +1,293 @@
+     */
+    write(path: any, data: object): Promise<any>;
+    private __setupLogger;
+}
+
+declare class VaultApiClient {
+    /**
+     * @param {Object} config
+     * @param {String} config.url - the url of the vault server
+     * @param {String} [config.apiVersion='v1']
+     * @param {Object} logger
+     */
+    constructor(config: {
+        url: string;
+        apiVersion?: string;
+    }, logger: any);
+    _logger: any;
+    makeRequest(method: any, path: any, data: any, headers: any): any;
+}
+
+declare class Lease {
+    static fromResponse(response: any): Lease;
+    constructor(requestId: any, leaseId: any, leaseDuration: any, renewable: any, data: any);
+    /**
+     * @param {String} key
+     * @returns {Object}
+     */
+    getValue(key: string): any;
+    /**
+     * @returns {Object}
+     */
+    getData(): any;
+    /**
+     * @returns {bool}
+     */
+    isRenewable(): boolean;
+}
+
+declare class VaultNodeConfig {
+    constructor(vault: any);
+    /**
+     * Populates Vault's values to "node-config"
+     */
+    populate(): any;
+}
+
+declare class VaultError extends Error {
+    constructor(message: any, error: any);
+    name: any;
+    message: any;
+}
+declare class InvalidArgumentsError extends VaultError {
+}
+declare class InvalidAWSCredentialsError extends InvalidArgumentsError {
+}
+declare class AuthTokenExpiredError extends VaultError {
+}
+
+declare class AuthToken {
+    static fromResponse(response: any): AuthToken;
+    /**
+     * @param {string} id
+     * @param {string} accessor
+     * @param {number} createdAt - UNIX timestamp
+     * @param {number|null} expiresAt - UNIX timestamp
+     * @param {number} explicitMaxTtl - in seconds
+     * @param {number} numUses
+     * @param {boolean} isRenewable
+     */
+    constructor(
+        id: string,
+        accessor: string,
+        createdAt: number,
+        expiresAt: number | null,
+        explicitMaxTtl: number,
+        numUses: number,
+        isRenewable: boolean,
+    );
+    /**
+     * @returns {string}
+     */
+    getId(): string;
+    getAccessor(): string;
+    /**
+     * @returns {boolean}
+     */
+    isExpired(): boolean;
+    /**
+     * @returns {boolean}
+     */
+    isRenewable(): boolean;
+    /**
+     * @returns {number|null} UNIX timestamp
+     */
+    getExpiresAt(): number | null;
+}
+
+declare class VaultAppRoleAuth extends VaultBaseAuth {
+    /**
+     * @param {VaultApiClient} apiClient - see {@link VaultBaseAuth#constructor}
+     * @param {Object} config
+     * @param {String} config.role_id - RoleID of the AppRole.
+     * @param {String} [config.secret_id] - required when bind_secret_id is enabled SecretID belonging to AppRole.
+     * @param {String} mount - Vault's  mount point ("approle" by default)
+     */
+    constructor(apiClient: VaultApiClient, logger: any, config: {
+        role_id: string;
+        secret_id?: string;
+    }, mount: string);
+    _authenticate(): any;
```

Comment:
Anything with an underscore is private, no? The API given in the package readme is much smaller than the types here (and elsewhere)

Reply:
Hi @jakebailey - single underscores are protected.

Reply:
But they're not documented, right? I don't see any evidence they need to be there...

Reply:
Hi @jakebailey - They are now removed.